src/test/eo/org/eolang/threads/mutex-test.eo

```diff
@@ -164,6 +161,37 @@
         t10.join
     $.equal-to TRUE
 
+[] > stress-test-2
```

Comment:
I think the test failed becuse of it's complexity for eo. More simple test without threads:
```
[] > stress-test-2
  [size] > list-by-size
    if. > @
      size.lte 0
      list *
      with.
        list-by-size
          size.minus 1
        TRUE
  list-by-size 9 > l
  assert-that > @
    l.reduced
      TRUE
      [a x]
        and. > @
          a
          TRUE
    $.equal-to TRUE
```
failes with the same trace:
```
[INFO] Running EOorg.EOeolang.EOthreads.EOstress_test_2Test
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.461 s <<< FAILURE! - in EOorg.EOeolang.EOthreads.EOstress_test_2Test
[ERROR] EOorg.EOeolang.EOthreads.EOstress_test_2Test.testWorks  Time elapsed: 2.452 s  <<< ERROR!
EOorg.EOeolang.EOerror$ExError: EOorg.EOeolang.EOstringν12912="Unexpected error 'null' of type StackOverflowError; caused by "SF
        at org.eolang.PhSafe.attr(PhSafe.java:97)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhSafe.attr(PhSafe.java:94)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhSafe.attr(PhSafe.java:94)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhImmovable.attr(PhImmovable.java:85)
        at org.eolang.PhLocated.attr(PhLocated.java:99)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhDefault.attr(PhDefault.java:256)
        at org.eolang.PhLocated.attr(PhLocated.java:99)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhSafe.attr(PhSafe.java:94)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhImmovable.attr(PhImmovable.java:85)
        at org.eolang.PhLocated.attr(PhLocated.java:99)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhDefault.attr(PhDefault.java:256)
        at org.eolang.PhLocated.attr(PhLocated.java:99)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhSafe.attr(PhSafe.java:94)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhImmovable.attr(PhImmovable.java:85)
        at org.eolang.PhLocated.attr(PhLocated.java:99)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhMethod.lambda$new$0(PhMethod.java:42)
        at org.eolang.Data$Once.lambda$take$0(Data.java:119)
        at java.base/java.util.concurrent.atomic.AtomicReference.updateAndGet(AtomicReference.java:209)
        at org.eolang.Data$Once.take(Data.java:115)
        at org.eolang.PhOnce.copy(PhOnce.java:81)
        at org.eolang.PhLocated.copy(PhLocated.java:84)
        at org.eolang.Data$Once.lambda$take$0(Data.java:119)
        at java.base/java.util.concurrent.atomic.AtomicReference.updateAndGet(AtomicReference.java:209)
        at org.eolang.Data$Once.take(Data.java:115)
        at org.eolang.PhOnce.attr(PhOnce.java:91)
        at org.eolang.PhWith.lambda$new$3(PhWith.java:86)
        at org.eolang.Data$Once.lambda$take$0(Data.java:119)
        at java.base/java.util.concurrent.atomic.AtomicReference.updateAndGet(AtomicReference.java:209)
        at org.eolang.Data$Once.take(Data.java:115)
        at org.eolang.PhOnce.attr(PhOnce.java:91)
        at org.eolang.PhWith.lambda$new$3(PhWith.java:86)
        at org.eolang.Data$Once.lambda$take$0(Data.java:119)
        at java.base/java.util.concurrent.atomic.AtomicReference.updateAndGet(AtomicReference.java:209)
        at org.eolang.Data$Once.take(Data.java:115)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhDefault.attr(PhDefault.java:256)
        at org.eolang.PhLocated.attr(PhLocated.java:99)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhDefault.attr(PhDefault.java:256)
        at org.eolang.Dataized.take(Dataized.java:87)
        at EOorg.EOeolang.EOthreads.EOstress_test_2Test.testWorks(EOstress_test_2Test.java:90)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
        at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
        at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
        at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
        at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:55)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:223)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:175)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:139)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)

```
So we need just decrease size of `list-by-size`.